File: pset6/credit.py
from cs50 import get_int

cardnumber = -1
while cardnumber<0:
    cardnumber = get_int("Number: ")

total = 0
ae = 0
digit = 0
mc = 0
visa = 0
i = 0

while cardnumber>0:
    digit += 1
    if i % 2 == 0:
        remainder = cardnumber % 10
        total += remainder
    else:
        remainder = 2*(cardnumber % 10)
        total += (remainder//10) + (remainder%10)
    cardnumber = cardnumber // 10
    if(cardnumber<100 and (cardnumber == 34 or cardnumber == 37)):
            ae = 1
    elif(cardnumber<100 and cardnumber <= 55 and cardnumber >= 51):
            mc = 1
    elif(cardnumber<10 and cardnumber == 4):
        visa = 1
    i += 1

if(total%10==0 and ae!=0):
    print("AMEX")
elif(total%10==0 and visa!=0 and digit>=13 and digit <=16):
    print("VISA")
elif(total%10==0 and mc!=0):
    print("MASTERCARD")
else:
    print("INVALID")

File: pset1/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main()
{
    float somadigito;
    int i=0, visa = 0, ae = 0, mc = 0, digito = 0;
    long novo, somaproduto=0, remainder, aux;
    
    long sif = get_long("Number: ");
    
    
    while(sif>0)
    {
        digito++;
        if(i%2==0)
        {
            remainder = fmodl(sif,10);
            somaproduto += remainder;
        }
        else
        {
            remainder=2*fmodl(sif,10);
            if(remainder<10)
            {
                somaproduto+= remainder;
            }
            else
            {
                aux = fmodl(remainder,10);
                remainder = floor(remainder/10);
                somaproduto+= aux + remainder;
            }
        }
        sif=sif/10;
        if(sif<100 && (sif == 34 || sif == 37))
        {
            ae++;
        }
        else if(sif<100 && sif <= 55 && sif >= 51)
        {
            mc++;
        }
        else if(sif<10 && sif == 4)
        {
            visa++;
        }
    i++;
    }
    
    if(somaproduto%10==0 && ae!=0)
    {
        printf("AMEX\n");
    }
    else if(somaproduto%10==0 && visa!=0 && digito>=13 && digito <=16)
    {
        printf("VISA\n");
    }
    else if(somaproduto%10==0 && mc!=0)
    {
        printf("MASTERCARD\n");
    }
    else printf("INVALID\n");
}
